jstests/core/distinct_multikey_index.js

```diff
@@ -0,0 +1,32 @@
+(function() {
+    "use strict";
+
+    load("jstests/libs/analyze_plan.js");
+
+    var t = db.distinct_multikey_index;
+
+    t.drop();
+    for (var i = 0; i < 10; i++) {
+        t.save({a: 1, b: 1});
+        t.save({a: 1, b: 2});
+        t.save({a: 2, b: 1});
+        t.save({a: 2, b: 3});
+    }
+    t.createIndex({a: 1, b: 1});
+
+    var explain = t.distinct('b', {a: 1}).explain("executionStats");
+    assert.commandWorked(explain);
+    assert(planHasStage(explain.queryPlanner.winningPlan, "DISTINCT_SCAN"));
+    assert(planHasState(explain.queryPlanner.winningPlan, "PROJECTION"));
+    assert.eq(2, explain.executionStats.nReturned);
+
```

Comment:
Perhaps also add a negative test showing that we don't use DISTINCT_SCAN when there is an empty query predicate.